src/server/handlers.rs

```diff
@@ -1075,6 +1078,61 @@ fn parse_image_ref(image_name: &str) -> Option<(String, String)> {
     }
 }
 
+/// This function validates that a key/value pair is a valid OAM label. It's using fairly
+/// basic validation rules to ensure that the manifest isn't doing anything horribly wrong. Keeping
+/// this function free of regex is intentional to keep this code functional but simple.
+///
+/// See <https://github.com/oam-dev/spec/blob/master/metadata.md#metadata> for details
+fn valid_oam_label(label: (&String, &String)) -> bool {
+    let (key, _) = label;
+    match key.split_once('/') {
+        Some((prefix, name)) => is_valid_dns_subdomain(prefix) && is_valid_label_value(name),
+        None => is_valid_label_value(key),
+    }
+}
+
+fn is_valid_dns_subdomain(s: &str) -> bool {
+    if s.is_empty() || s.len() > 253 {
+        return false;
+    }
+
+    s.split('.').all(|part| {
+        !part.is_empty()
+            && part.len() <= 63
+            && part.chars().all(|c| c.is_ascii_alphanumeric() || c == '-')
+            && part
```

Comment:
Are these last two checking for leading and trailing whitespace? Would it just be clearer to use `starts_with` and `ends_with` instead?

Reply:
It's actually checking to see if the first and last character is an alphanumeric character

Reply:
Will check out using the starts_with with a pattern to make this better. Also I can leave a comment here to say why I'm checking